app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,29 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    puts 'Current User'
+
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    respond_to do |_format|
+      if @post.save
+        redirect_to user_posts_path(current_user)
```

Comment:
- According to the project requirements, when a new post is created it should redirect to the `user_post_path`, this is not so in your project, please fix this 🙏